### PR TITLE
feat: configurable DB port

### PR DESCRIPTION
**Pull request by mattvella07**
Added configurable DB port as mentioned in #198 but unsure about the tests, some are failing.  Also wasn't sure about adding the port to the Azure MySQL database type since that isn't configurable currently

### CODE REVIEW

File: api/proto/resourcespb/database.proto [hunk 1]
```diff
@@ -53,8 +54,9 @@ message DatabaseResource {
   string username = 7;
   string password = 8;
   repeated string subnet_ids = 9;
+  int32 port = 10;
```
Comment (goncalo-rodrigues) [inline api/proto/resourcespb/database.proto:57]: Just use 12 here.

In protobufs, if you change the number of existing values (host and connection_username), you lose backwards compability. This explains better how protbuf encoding works: https://developers.google.com/protocol-buffers/docs/encoding#structure

File: resources/output/database/azure_databases.go [hunk 2]
```diff
@@ -16,6 +17,7 @@ type AzureDbServer struct {
 	Version                    string
 	Engine                     string
 	SubnetIds                  []string
+	Port                       int
```
Comment (goncalo-rodrigues) [inline resources/output/database/azure_databases.go:20]: you can take this out since azure doesn't let you configure it

**Comment by goncalo-rodrigues** [comment]
You're right, azure isn't configurable as of right now. Please return a validation error in the database's Validate method if the port is set (i.e. > 0) and cloud is azure.

As for the tests, it prints what it expects and what was actually generated. You can run them locally, for example to run the one that is failing, you can do:
go test -v ./test/... -run Translate/subnet_public_private

It prints 2 things right now:
1) The code that was generated by multy
2) The diffs between the generated code and the expected (main.tf)

**Comment by JCoelhoo** [comment]
This parameter will become an AWS override in the future (https://docs.multy.dev/overrides)

**Comment by mattvella07** [comment]
@goncalo-rodrigues Thanks for the comments.  I'm still getting an error on this test TestConfigTranslate/subnet_public_private and I was wondering if its because the terraform project hasn't been updated to accept the `port` parameter

```
 unexpected attribute 'port' 
        20:   port                 = 0
--- FAIL: TestConfigTranslate (0.04s)
    --- FAIL: TestConfigTranslate/subnet_public_private (0.04s)
FAIL
FAIL	github.com/multycloud/multy/test	0.210s
testing: warning: no tests to run
```

**Comment by goncalo-rodrigues** [comment]
Thanks for the contribution!

What's happening is that there's a default value of 0 for the port, and when encoding into HCL, you're encoding that 0. So the generated file in that test has an attribute in the database saying "port = 0", but the expected main.tf doesn't (correctly), so you get that error.

What you should do is to not encode the default value, so just add hcle:"omitempty" to the port field in AwsDbInstance. That will ignore the default value and not encode it. We do that all the time when we want to ignore empty strings and things like that. For example:
https://github.com/multycloud/multy/blob/c0c83580761a54e99d9d9eddd43eca43b6b1428e/resources/output/object_storage_object/aws_object_storage_object.go

**Comment by mattvella07** [comment]
@goncalo-rodrigues That makes sense, thanks for the explanation!

**Comment by goncalo-rodrigues** [comment]
Awesome, thanks for the great work!

**Pull request merged**